index.js

```diff
@@ -1,59 +1,37 @@
 'use strict';
 
-const ANNOTATION_EXECUTOR_TYPE = 'beta.screwdriver.cd/executor'; // Key in annotations object that maps to an executor NPM module
 const Executor = require('screwdriver-executor-base');
+const Resque = require('node-resque');
+const fuses = require('circuit-fuses');
+const Breaker = fuses.breaker;
+const Fusebox = fuses.box;
 
-class ExecutorRouter extends Executor {
+class ExecutorQueue extends Executor {
     /**
      * Constructs a router for different Executor strategies.
      * @method constructor
      * @param  {Object}         config                      Object with executor and ecosystem
-     * @param  {String}         [config.defaultPlugin]      Optional default executor
-     * @param  {Object}         [config.ecosystem]          Optional object with ecosystem values
-     * @param  {Array|Object}   config.executor             Array of executors to load or a single executor object
-     * @param  {String}         config.executor[x].name     Name of the executor NPM module to load
-     * @param  {String}         config.executor[x].options  Configuration to construct the module with
+     * @param  {Object}         config.redisConnection      Connection details for redis
      */
     constructor(config = {}) {
-        const ecosystem = config.ecosystem || {};
-        const executorConfig = config.executor;
-        const defaultPlugin = config.defaultPlugin;
-
-        if (!executorConfig || !(Array.isArray(executorConfig)) || executorConfig.length === 0) {
-            throw new Error('No executor config passed in.');
+        if (!config.redisConnection) {
+            throw new Error('No redis connection passed in');
         }
 
         super();
 
-        executorConfig.forEach((plugin) => {
-            let ExecutorPlugin;
-
-            try {
-                // eslint-disable-next-line global-require, import/no-dynamic-require
-                ExecutorPlugin = require(`screwdriver-executor-${plugin.name}`);
-            } catch (err) {
-                console.error(err);
-
-                return;
-            }
+        const redisConnection = Object.assign(config.redisConnection, { pkg: 'ioredis' });
```

Comment:
this will mutate `config.redisConnection`, you could do `Object.assign({}, config.redisConnection, { pkg: 'ioredis' });` if that's not intended.